frontend/app/component/election/ElectionStatusProgress.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+
+import { PollingStationStatus, PollingStationStatusEntry, useElection, useElectionStatus } from "@kiesraad/api";
+import { IconDot } from "@kiesraad/icon";
+import { Icon, PercentageAndColorClass, Progress, ProgressBar, ProgressBarColorClass } from "@kiesraad/ui";
+
+const statusCategories = ["in_progress", "unfinished", "definitive", "not_started"] as const;
+type StatusCategory = (typeof statusCategories)[number];
+
+const categoryTitle: Record<StatusCategory, string> = {
+  unfinished: "Niet afgeronde invoer",
+  in_progress: "Invoer bezig",
+  definitive: "Eerste invoer klaar", // TODO: in the future this changes to `Eerste en tweede invoer klaar`
+  not_started: "Werkvoorraad",
+};
+
+const categoryColorClass: Record<StatusCategory, ProgressBarColorClass> = {
+  unfinished: "unfinished",
+  in_progress: "in-progress",
+  definitive: "definitive",
+  not_started: "not-started",
+};
+
+function statusCount(entries: PollingStationStatusEntry[], status: PollingStationStatus): number {
+  return entries.filter((s) => s.status === status).length;
+}
+
+export function ElectionStatusProgress() {
+  const { pollingStations } = useElection();
+  const { statuses } = useElectionStatus();
+
+  const categoryCounts: Record<StatusCategory, number> = React.useMemo(() => {
+    // TODO: future `second_entry_unfinished` status should be added to `unfinished`
+    //  future `second_entry_in_progress` status should be added to `in_progress`
+    return {
+      unfinished: statusCount(statuses, "first_entry_unfinished"),
+      in_progress: statusCount(statuses, "first_entry_in_progress"),
+      definitive: statusCount(statuses, "definitive"),
+      not_started: statusCount(statuses, "first_entry"),
+    };
+  }, [statuses]);
+
+  const progressBarData: PercentageAndColorClass[] = React.useMemo(() => {
+    const total = pollingStations.length;
+    return statusCategories.map((cat) => ({
+      percentage: Math.round(categoryCounts[cat] / total) * 100,
```

Comment:
If there are no polling stations, `total` is `0` and `percentage` becomes `NaN`.

Having zero polling stations also causes the ElectionStatusPage and the DataEntryHomePage to display "Alle stembureaus zijn twee keer ingevoerd" / "Alle stembureaus zijn ingevoerd", since they do an `every()` over an empty array.
I considered if this is fine for the data entry page. However, it shows an error in the list of polling stations, while if data for all polling stations has been submitted, it shows an empty table. So we do make the distinction between zero polling stations for the election and zero polling stations needing data entry.

Having no polling stations also causes `/elections/1/data-entry` to show `NaN%` for "Voortgang - Alles samen".

Reply:
Maybe we should prevent users to navigate to this page in case no pollingstations have been linked to the election (or more specific / in the future when we support it: _Zitting_ (first and following _zittingen_  can have a different selection of available polling stations). If there are no polling stations, then it is impossible to enter any data, because election data is always linked to a polling station. 

Proposed solution would be to disallow starting the data entry phase without at least 1 polling station selected.

Reply:
Fixed in ff94f4e9f3e410517a27e412c9daf77bb106e2c3